src/controllers/productsController.ts

```diff
@@ -0,0 +1,99 @@
+import { Request, Response } from 'express';
+import logger from '../logs/config';
+import uploadImage from '../helpers/claudinary';
+import { validateFields } from '../validations';
+import { Size, SizeCreationAttributes } from '../database/models/Size';
+import { Product, ProductAttributes } from '../database/models/Product';
+
+export const createProduct = async (req: Request, res: Response) => {
+  try {
+    // when products exists
+    const name = req.body.name;
+    const thisProductExists = await Product.findOne({ where: { name } });
+
+    if (thisProductExists) {
+      return res.status(400).json({
+        ok: false,
+        message: 'This Product already exists, You can update the stock levels instead.',
+        data: thisProductExists,
+      });
+    }
+    // handle images
+    const productImages = [];
+    const images: unknown = req.files;
+    if (images instanceof Array && images.length > 3) {
+      for (const image of images) {
+        const imageBuffer: Buffer = image.buffer;
```

Comment:
it's good practice using variables, however passing image.buffer would be sufficient, without declaring variables which reserves a space on storage which uncessary

Reply:
guess you're seeing that's an array of objects I'm looping thru and after that I grab the buffer property from the object, and lastly I ensure type safety